runtime/etc/Makefile.comm-ofi

```diff
@@ -27,6 +34,23 @@ LIBS += -lfabric -ldl
 LD = $(CXX)
 
 ifneq (, $(findstring mpi,$(CHPL_MAKE_LAUNCHER)))
-GEN_LFLAGS += -L$(MPI_LIB) -Wl,-rpath,$(MPI_LIB)
-LIBS += -lmpi
+  ifneq ($(MPI_DIR),)
+    MPI_DIR_CACHE := $(shell if [ -d $(MPI_DIR)/lib64 ] ; then \
+                               echo $(MPI_DIR)/lib64 ; \
+                             elif [ -d $(MPI_DIR)/lib ] ; then \
+                               echo $(MPI_DIR)/lib ; \
+                             fi)
+    ifneq ($(MPI_DIR_CACHE),)
+      GEN_LFLAGS += -L$(MPI_DIR_CACHE) -Wl,-rpath,$(MPI_DIR_CACHE)
+      LIBMPI_CACHE := \
+        $(shell if $$(ls $(MPI_DIR_CACHE)/libmpi.* >&/dev/null) ; then \
```

Comment:
Ditto comments about using wildcard. I think that would clean this logic up a lot